filter-api/src/main/java/org/hypertrace/agent/filter/FilterRegistry.java

```diff
@@ -81,12 +83,16 @@ private static Filter load(List<String> jarPaths) {
     ClassLoader cl = loadJars(jarPaths);
     ServiceLoader<FilterProvider> providers = ServiceLoader.load(FilterProvider.class, cl);
     List<Filter> filters = new ArrayList<>();
+
+    String serviceName = HypertraceConfig.get().getServiceName();
```

Comment:
Instead of this, we can modify the initialize() method to take in a FilterProviderConfig object and then at https://github.com/hypertrace/javaagent/blob/dd21236df53b29b77e80ad169b022ed5142a2bc8/otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/FilterComponentInstaller.java#L41 create it and invoke initialize with it. Right?

Reply:
Done.